tests/components/home_plus_control/test_init.py

```diff
@@ -75,11 +90,32 @@ async def test_loading(hass, mock_config_entry, current_request_with_host):
     assert mock_config_entry.state == config_entries.ENTRY_STATE_LOADED
 
 
+async def test_loading_with_no_config(
+    hass, mock_config_entry, current_request_with_host
+):
+    """Test component loading failure when it has not configuration."""
+    await setup.async_setup_component(hass, "http", {})
+    assert hass.http.app
+    await setup.async_setup_component(hass, DOMAIN, {})
+    await hass.config_entries.async_add(mock_config_entry)
+    assert mock_config_entry.state == config_entries.ENTRY_STATE_NOT_LOADED
+
+
 async def test_unloading(hass, mock_config_entry, current_request_with_host):
     """Test component unloading."""
     await setup.async_setup_component(hass, "http", {})
```

Comment:
We don't need to set up the http integration explicitly since we depend on it from this integration.